### PR TITLE
Refactoring of the basic client module

### PR DESCRIPTION
**Pull request by hendrikebbers**
- final
- null checks
- comment code removed
- TODOs that should be discussed

### CODE REVIEW

**Comment by hendrikebbers** [comment]
Maybe this has some duplicate changes with #23 that should be merged before this PR is merged. #23 can be merged once @joha78 reacted in then small review requests I did earlier today.

**Comment by madmas** [comment]
Overall, looks good to me and I agree that #23 needs to be merged first.*
*But* please  remove all Todo comments from the code before merge, either we should solve them within the PR or transfer each topic into an issue

File: base/client/src/main/java/dev/rico/internal/client/projector/JavaFXProjectorImpl.java
Comment (joha78) [inline base/client/src/main/java/dev/rico/internal/client/projector/JavaFXProjectorImpl.java]: the return null is for this case:
```
BorderPaneModel borderPaneModel = ui().borderPane();
        borderPaneModel.setCenter(something);
        borderPaneModel.setCenter(null);
```

File: base/client/src/main/java/dev/rico/internal/client/projector/dialoghandler/DialogConfiguration.java
Comment (joha78) [inline base/client/src/main/java/dev/rico/internal/client/projector/dialoghandler/DialogConfiguration.java]: Can be removed

File: base/client/src/main/java/dev/rico/internal/client/projector/dialoghandler/ShutdownDialogHandler.java
Comment (joha78) [inline base/client/src/main/java/dev/rico/internal/client/projector/dialoghandler/ShutdownDialogHandler.java]: will be removed by @joha78

File: base/client/src/main/java/dev/rico/internal/client/projector/mixed/Configuration.java
Comment (joha78) [inline base/client/src/main/java/dev/rico/internal/client/projector/mixed/Configuration.java]: Will be removed by @joha78

File: base/client/src/main/java/dev/rico/internal/client/projector/uimanager/AsyncSequence.java
Comment (joha78) [inline base/client/src/main/java/dev/rico/internal/client/projector/uimanager/AsyncSequence.java]: Will be removed by @joha78

File: base/client/src/main/java/dev/rico/internal/client/projector/uimanager/ImageViewer.java
Comment (joha78) [inline base/client/src/main/java/dev/rico/internal/client/projector/uimanager/ImageViewer.java]: Will be removed by @joha78

File: base/client/src/main/java/dev/rico/internal/client/projector/uimanager/UnexpectedErrorDialog.java
Comment (joha78) [inline base/client/src/main/java/dev/rico/internal/client/projector/uimanager/UnexpectedErrorDialog.java:98]: Messages should be configurable

**Review by joha78 — CHANGES_REQUESTED** [review]

**Review by madmas — CHANGES_REQUESTED** [review]
* remove TODO (convert into issues or solve)
* solve open discussions 

;-)

**Pull request merged**
